http-client-jdk/src/main/java/io/micronaut/http/client/jdk/AbstractJdkHttpClient.java

```diff
@@ -312,4 +343,64 @@ private <I> Mono<URI> resolveURI(io.micronaut.http.HttpRequest<I> request) {
     protected <O> HttpResponse<O> response(@NonNull java.net.http.HttpResponse<byte[]> netResponse, @NonNull Argument<O> bodyType) {
         return new HttpResponseAdapter<>(netResponse, bodyType, conversionService, mediaTypeCodecRegistry);
     }
+
+    protected <I, O> Flux<HttpResponse<O>> exchangeImpl(@NonNull io.micronaut.http.HttpRequest<I> request, @NonNull Argument<O> bodyType) {
+        var defaultPublisher = responsePublisher(request, bodyType);
+        return resolveRequestUri(request)
+            .flatMapMany(uri -> applyFilterToResponsePublisher(request, uri, defaultPublisher));
+    }
+
+    protected <I, R extends io.micronaut.http.HttpResponse<?>> Publisher<R> applyFilterToResponsePublisher(
+        io.micronaut.http.HttpRequest<I> request,
+        URI requestURI,
+        Publisher<R> responsePublisher
+    ) {
+        if (!(request instanceof MutableHttpRequest<?> mutRequest) || filterResolver == null) {
+            return responsePublisher;
+        }
+
+        mutRequest.uri(requestURI);
+        List<GenericHttpFilter> filters =
+            filterResolver.resolveFilters(request, clientFilterEntries);
+
+        FilterRunner.sortReverse(filters);
+        filters.add(new GenericHttpFilter.TerminalReactive(responsePublisher));
+
+        FilterRunner runner = new FilterRunner(filters);
+        return Mono.from(ReactiveExecutionFlow.fromFlow((ExecutionFlow<R>) runner.run(request)).toPublisher());
+    }
+
+    protected <O> Publisher<io.micronaut.http.HttpResponse<O>> responsePublisher(
+        io.micronaut.http.HttpRequest<?> request,
+        Argument<O> bodyType
+    ) {
+        return Flux.defer(() -> mapToHttpRequest(request, bodyType)) // defered so any client filter changes are used
+            .map(httpRequest -> {
+                if (log.isDebugEnabled()) {
+                    log.debug("Client {} Sending HTTP Request: {}", clientId, httpRequest);
+                }
+                if (log.isTraceEnabled()) {
+                    httpRequest.headers().map().forEach((k, v) -> log.trace("Client {} Sending HTTP Request Header: {}={}", clientId, k, v));
```

Comment:
let's use `HttpHeaderUtils` to take into account logging `HEADER_MASK_PATTERNS`

Reply:
I created a PR https://github.com/micronaut-projects/micronaut-core/pull/9715

Reply:
👍